From 84cec998b9485e1107ade5390230cbdb6729d8dc Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Mon, 25 Sep 2017 12:01:05 +0300
Subject: [PATCH 125/767] staging: fsl-dpaa2/eth: Field rename

Rename a couple of fields in the channel structure, as the
current name is very similar to another and may cause confusion.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../staging/fsl-dpaa2/ethernet/dpaa2-eth.c    | 20 ++++++++++++-------
 .../staging/fsl-dpaa2/ethernet/dpaa2-eth.h    |  4 ++--
 2 files changed, 15 insertions(+), 9 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
index 698fe3f999f7..1ba47128e142 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
@@ -283,15 +283,21 @@ static void release_fd_buf(struct dpaa2_eth_priv *priv,
 			   struct dpaa2_eth_channel *ch,
 			   dma_addr_t addr)
 {
-	ch->buf_array[ch->buf_cnt++] = addr;
-	if (likely(ch->buf_cnt < DPAA2_ETH_BUFS_PER_CMD))
+	ch->rel_buf_array[ch->rel_buf_cnt++] = addr;
+	if (likely(ch->rel_buf_cnt < DPAA2_ETH_BUFS_PER_CMD))
 		return;
 
-	while (dpaa2_io_service_release(NULL, priv->bpid, ch->buf_array,
-					ch->buf_cnt))
-		cpu_relax();
-
-	ch->buf_cnt = 0;
+ 	do {
+		err = dpaa2_io_service_release(NULL, priv->bpid,
+					       ch->rel_buf_array,
+					       ch->rel_buf_cnt);
+ 		cpu_relax();
+ 	} while (err == -EBUSY);
+ 
+ 	if (unlikely(err))
+		free_bufs(priv, ch->rel_buf_array, ch->rel_buf_cnt);
+ 
+	ch->rel_buf_cnt = 0;
 }
 
 /* Main Rx frame processing routine */
diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
index 71cbc83df02b..c50122dd8c13 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
@@ -355,8 +355,8 @@ struct dpaa2_eth_channel {
 	int buf_count;
 	struct dpaa2_eth_ch_stats stats;
 	struct bpf_prog *xdp_prog;
-	u64 buf_array[DPAA2_ETH_BUFS_PER_CMD];
-	u8 buf_cnt;
+	u64 rel_buf_array[DPAA2_ETH_BUFS_PER_CMD];
+	u8 rel_buf_cnt;
 };
 
 struct dpaa2_eth_cls_rule {
-- 
2.17.0

